src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Globalization;
+using System.Linq;
+using Datadog.Trace.Headers;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace
+{
+    internal class SpanContextPropagator
+    {
+        private const NumberStyles NumberStyles = System.Globalization.NumberStyles.Integer;
+
+        private static readonly CultureInfo InvariantCulture = CultureInfo.InvariantCulture;
+        private static readonly ILog Log = LogProvider.For<SpanContextPropagator>();
+
+        private SpanContextPropagator()
+        {
+        }
+
+        public static SpanContextPropagator Instance { get; } = new SpanContextPropagator();
+
+        /// <summary>
+        /// Propagates the specified context by adding new headers to a <see cref="IHeadersCollection"/>.
+        /// This locks the sampling priority for <paramref name="context"/>.
+        /// </summary>
+        /// <param name="context">A <see cref="SpanContext"/> value that will be propagated into <paramref name="headers"/>.</param>
+        /// <param name="headers">A <see cref="IHeadersCollection"/> to add new headers to.</param>
+        public void Inject(SpanContext context, IHeadersCollection headers)
+        {
+            if (context == null) { throw new ArgumentNullException(nameof(context)); }
+
+            if (headers == null) { throw new ArgumentNullException(nameof(headers)); }
+
+            headers.Set(HttpHeaderNames.TraceId, context.TraceId.ToString(InvariantCulture));
+            headers.Set(HttpHeaderNames.ParentId, context.ParentId?.ToString(InvariantCulture));
```

Comment:
This should be the current spans id, not the trace's parent id.

```
Span A
  | Span B
    | -> Http request
      | X-Datadog-Trace-Id: Span B.TraceContext.TraceId (seems like Span B.TraceId is fine too)
      | X-Datadog-Parent-Id: Span B.SpanId
      | X-Datadog-Sampling-Priority: Span B.TraceContext.SamplingPriority
```

This way when this gets picked up by another service on the other end we know to group it's spans under `Span B`

Reply:
Good catch. Fixed in 03fc05360d96609ca1c93705f7aaf26083455a99.